src/solvers/refinement/string_constraint_generator.h

```diff
@@ -320,14 +320,24 @@ class string_constraint_generatort: messaget
   exprt add_axioms_for_offset_by_code_point(
     const function_application_exprt &f);
 
-  exprt add_axioms_for_characters_in_integer_string(
-    const symbol_exprt& x,
+  void add_axioms_for_characters_in_integer_string_nonconstructive(
+    const exprt& x,
     const typet& type,
     const typet& char_type,
     const typet& index_type,
     const string_exprt& str,
     const std::size_t max_string_length,
-    const exprt& radix);
+    const exprt& radix,
+    const unsigned long radix_ul=0);
```

Comment:
This is a lot of arguments. You could recover `char_type` and `index_type` from the type of `str` so they are redundant.